### PR TITLE
CMR-6451: Autocomplete permission/group filtering

### PR DESCRIPTION
**Pull request by mschmele**
Update autocomplete index and search logic to support filtering results to users based on associated collection permissions

### CODE REVIEW

File: indexer-app/src/cmr/indexer/services/index_service.clj [hunk 1]
```diff
@@ -205,46 +195,63 @@
...
-                           (s/replace #"([_\.]?humanized(_?2)?|-sn|-id)" ""))
-              value-map (as-> humanized-field h
-                          (val h)
-                          (map util/remove-nil-keys h)
-                          (map #(dissoc % :priority) h))
-              suggestion-docs (map #(suggestion-doc key-name %)
-                                   value-map)]]
-    suggestion-docs))
+  (let [{:keys [permissions]} humanized-fields
+        fields-without-permissions (-> humanized-fields
```
Comment (ygliuvt) [inline indexer-app/src/cmr/indexer/services/index_service.clj:240]: change the value to `(dissoc humanized-fields :id :permissions)`

File: search-app/src/cmr/search/services/autocomplete_service.clj [hunk 2]
```diff
@@ -1,25 +1,62 @@
...
+
+(defn- user-groups-condition
+  [user-acls]
+  (if-not (empty? user-acls)
+    (gc/or-conds [public-collections-condition
+                  (gc/or-conds
+                   (map #(qm/text-condition :permitted-group-ids %) user-acls))])
+    public-collections-condition))
+
+(defn- empty-token-with-type
```
Comment (ygliuvt) [inline search-app/src/cmr/search/services/autocomplete_service.clj:20]: Add docstring to these functions.

File: indexer-app/src/cmr/indexer/services/index_service.clj (hunk 1 above)
Comment (mschmele) [reply indexer-app/src/cmr/indexer/services/index_service.clj]: Fixed - good catch

File: search-app/src/cmr/search/services/autocomplete_service.clj (hunk 2 above)
Comment (mschmele) [reply search-app/src/cmr/search/services/autocomplete_service.clj:22]: Done

**Pull request merged**